label_studio/core/utils/io.py

```diff
@@ -140,9 +140,14 @@ def read_yaml(filepath):
     return data
 
 
-def read_bytes_stream(filepath):
-    with open(filepath, mode='rb') as f:
-        return io.BytesIO(f.read())
+def path_to_open_binary_file(filepath) -> io.BufferedReader:
+    """
+    The result of this call should generally be passed to a FileResponse or similar.
+    Unusually, this call deliberately doesn't close the file; FileResponse is responsible for this.
+    """
+    tmp = NamedTemporaryFile(delete=True)
+    shutil.copy2(filepath, tmp.name)
```

Comment:
what does it do? why do we need it?

Reply:
If I just `open` the file at `filepath` here, it gets cleaned up before `FileResponse` finishes using the file. Instead I'm just copying the tempfile to the `NamedTemporaryFile`, which exists until it's closed (by FileResponse, or the caller).